types/chai/chai-tests.ts

```diff
@@ -1470,9 +1517,38 @@ suite('assert', () => {
     });
 
     test('instanceOf', () => {
-        assert.instanceOf(new Foo(), Foo);
-        assert.instanceOf(5, Foo);
-        assert.instanceOf(new CrashyObject(), CrashyObject);
+        {
+            const value = new Foo();
+            assert.instanceOf(value, Foo);
+
+            // $ExpectType Foo
+            value;
+        }
+
+        {
+            // tslint:disable-next-line:no-construct
+            const value = new Number(5);
+            assert.instanceOf(5, Number);
+
+            // $ExpectType Number
+            value;
+        }
+
+        {
+            const value = new CrashyObject();
+            assert.instanceOf(value, CrashyObject);
+
+            // $ExpectType CrashyObject
+            value;
+        }
+
+        {
+            const value = 5 as number;
+            assert.instanceOf(5, Foo);
+
+            // $ExpectType never
+            value;
+        }
```

Comment:
I am stumped on this failure; for the life of me, I cannot figure out what's wrong with the [type predicate for `instanceOf`](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64954/files#diff-a20309f5f243674900ddcb8811b893d1d2000c25c60199ae80ccfa48aaba9984R830) and why it's failing. An extra set of eyes would be greatly appreciated

```
Error: /home/vsts/work/1/s/types/chai/chai-tests.ts:1550:13
ERROR: 1550:13  expect  TypeScript@5.1 expected type to be:
  never
got:
  number

    at testTypesVersion (/home/vsts/work/1/s/node_modules/@definitelytyped/dtslint/src/index.ts:235:11)
    at runTests (/home/vsts/work/1/s/node_modules/@definitelytyped/dtslint/src/index.ts:197:7)
```